datalake/src/main/java/com/sequenceiq/datalake/service/sdx/CloudbreakFlowService.java

```diff
@@ -35,18 +35,30 @@ public void getAndSaveLastCloudbreakFlowChainId(SdxCluster sdxCluster) {
         sdxClusterRepository.save(sdxCluster);
     }
 
-    public boolean isLastKnownFlowRunning(SdxCluster sdxCluster) {
+    public FlowState getLastKnownFlowState(SdxCluster sdxCluster) {
         try {
             String actualCbFlowChainId = sdxCluster.getLastCbFlowChainId();
             if (actualCbFlowChainId != null) {
-                return flowEndpoint.hasFlowRunningByChainId(sdxCluster.getLastCbFlowChainId()).getHasActiveFlow();
+                LOGGER.info("Check if flow is running: {}", actualCbFlowChainId);
+                Boolean hasActiveFlow = flowEndpoint.hasFlowRunningByChainId(sdxCluster.getLastCbFlowChainId()).getHasActiveFlow();
+                if (hasActiveFlow) {
+                    return FlowState.RUNNING;
+                } else {
+                    return FlowState.FINISHED;
+                }
+            } else  {
+                return FlowState.UNKNOWN;
             }
         } catch (NotFoundException e) {
             LOGGER.error("Flow chain id or resource {} not found in CB: {}, so there is no active flow!", sdxCluster.getClusterName(), e.getMessage());
+            return FlowState.UNKNOWN;
         } catch (Exception e) {
             LOGGER.error("Exception occured during checking if there is a flow for cluster {} in CB: {}", sdxCluster.getClusterName(), e.getMessage());
-            return true;
+            return FlowState.UNKNOWN;
         }
-        return false;
+    }
+
+    public enum FlowState {
```

Comment:
What's wrong with StateStatus? Why introduce a new Enum?

Reply:
I would like to handle unknown state like missing flow id or flow id is not known by cloudbreak. Also we call hasFlowRunningByChainId method which can only return a boolean.